Source/SwiftLintCore/Extensions/SwiftLintFile+Regex.swift

```diff
@@ -273,6 +273,7 @@ extension SwiftLintFile {
         return violatingRanges.filter { range in
             let region = fileRegions.first {
                 $0.contains(Location(file: self, characterOffset: range.location))
+                || $0.contains(Location(file: self, characterOffset: range.location + range.length - 1))
```

Comment:
While this seems like a reasonable change, it has the potential of changing other corrections of rules as well (which we don't see in the OSS report). I'd rather like to avoid that. This will anyway become a thing if the past once these rules are converted to SwiftSyntax rules.

Actually, the decision if a rule is enabled or not should be the same for violations and corrections. In your example, a violations is reported, but no fix is applied. That's inconsistent. So we have to make sure that the "Is rule enabled?" check is implemented in the same way or at least can rely on the same positions in the "reporting" and "fix" cases, probably by extending `ViolationCorrection` with a `violationPosition`. Later, `ReasonedRuleViolation` and `ViolationCorrection` can even be merged.

Reply:
This has been addressed by #5680 and #5682.